packages/filesystem/src/browser/file-tree/file-tree-widget.ts

```diff
@@ -11,18 +11,27 @@ import {
     ContextMenuRenderer, VirtualRenderer,
     TreeWidget, NodeProps, TreeProps, ITreeNode
 } from "@theia/core/lib/browser";
+import { ElementAttrs } from "@phosphor/virtualdom";
 import { DirNode, FileStatNode } from "./file-tree";
 import { FileTreeModel } from "./file-tree-model";
+import { FileSystem } from '../../common';
+import * as base64 from 'base64-js';
+import { ConfirmDialog } from "@theia/core/lib/browser";
+import { ILogger } from '@theia/core/lib/common';
 
 export const FILE_TREE_CLASS = 'theia-FileTree';
 export const FILE_STAT_NODE_CLASS = 'theia-FileStatNode';
 export const DIR_NODE_CLASS = 'theia-DirNode';
 export const FILE_STAT_ICON_CLASS = 'theia-FileStatIcon';
+export const DROPZONE_CLASS = 'theia-Dropzone';
+const activeDropZone = "activeDropZone";
 
 @injectable()
 export class FileTreeWidget extends TreeWidget {
 
     constructor(
+        @inject(ILogger) protected readonly logger: ILogger,
+        @inject(FileSystem) protected readonly fileSystem: FileSystem,
```

Comment:
`FileSystem` is already injected in `FileTreeModel`, you should add upload feature `FileTreeModel` and call it from here, instead of injecting `FileSystem` here. Widget responsibility is rendering, not performing operations.

Reply:
Ok, created the upload method in FileTreeModel